src/cmap/mod.rs

```diff
@@ -405,13 +380,15 @@ impl Drop for ConnectionPoolInner {
     /// `ConnectionPoolInner` are dropped.
     fn drop(&mut self) {
         let address = self.address.clone();
-        let connection_manager = self.connection_manager.clone();
+        let available_connections =
+            std::mem::replace(&mut self.available_connections, Mutex::new(Vec::new()));
         let event_handler = self.event_handler.clone();
 
         RUNTIME.execute(async move {
-            let mut connection_manager = connection_manager.lock().await;
-            while let Some(connection) = connection_manager.checked_in_connections.pop() {
-                connection_manager.close_connection(connection, ConnectionClosedReason::PoolClosed);
+            // this lock attempt will always succeed.
```

Comment:
Can we add "immediately" or something to this comment to indicate that the lock won't be held by anything else? "succeed" alone sounds like it's staying it won't return an error, which is evident from looking at it

Reply:
hah, good point, fixed.